File: BackupUtility/Classes/ExceptionHandling/ContinueOnExceptionHandler.vb
Imports ZCopy

Public Class ContinueOnExceptionHandler
    Implements IExceptionHandler
    Private _commandHandler As CommandHandler

    Public Sub New(commandHandler As CommandHandler)
        _commandHandler = commandHandler
    End Sub

    Public Sub HandleException(message As String, ex As Exception) Implements IExceptionHandler.HandleException
        Dim eventText As String = message
        If (eventText = "" OrElse eventText Is Nothing) Then
            eventText = ex.Message
        End If
        _commandHandler.RaiseThisEvent(eventText)
    End Sub
End Class

File: BackupUtility/Classes/ExceptionHandling/ContinueOnExceptionHandler.cs
using GenericClassLibrary.Logging;
using System;
using ZCopy.Interfaces;

namespace ZCopy.Classes.ExceptionHandling
{
    public class ContinueOnExceptionHandler : IExceptionHandler
    {
        private readonly IEventLogger _eventLogger;

        public ContinueOnExceptionHandler(IEventLogger commandHandler)
        {
            _eventLogger = commandHandler;
        }

        public void HandleException(string message, Exception ex)
        {
            string eventText = message;
            if ((eventText == "" || eventText == null))
            {
                eventText = ex.Message;
            }
            Logger.Error($"Exception (IGNORED): {ex.Message}");
            _eventLogger.LogEvent(eventText);
        }
    }
}
